frontend/public/components/utils/use-url-hook.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useCallback, useState } from 'react';
+import { usePoll, useSafeFetch } from '.';
+
+export const URL_POLL_DEFAULT_DELAY = 15000; // 15 seconds
+
+export const useURLPoll = (
+  url: string,
+  delay = URL_POLL_DEFAULT_DELAY,
+  ...dependencies: any[]
+): URLPoll => {
+  const [error, setError] = useState();
+  const [response, setResponse] = useState();
+  const [loading, setLoading] = useState(true);
+  const safeFetch = useSafeFetch();
+  const tick = useCallback(() => {
+    if (url) {
+      safeFetch(url)
+        .then((data) => {
+          setResponse(data);
+          setError(undefined);
+          setLoading(false);
+        })
+        .catch((err) => {
+          if (err.name !== 'AbortError') {
+            setError(err);
+            setLoading(false);
+            // eslint-disable-next-line no-console
+            console.error(`Error polling URL: ${err}`);
+          }
+        });
+    } else {
+      setLoading(false);
+    }
+  }, [url]);
+
+  usePoll(tick, delay, ...dependencies);
+
+  return { response, error, loading };
```

Comment:
I'd prefer to return an array to align with other hooks
```suggestion
  return [ response, error, loading ];
```

Reply:
+1